pkg/services/sqlstore/transactions.go

```diff
@@ -4,63 +4,34 @@ import (
 	"context"
 	"time"
 
+	"github.com/go-xorm/xorm"
 	"github.com/grafana/grafana/pkg/bus"
 	"github.com/grafana/grafana/pkg/infra/log"
 	sqlite3 "github.com/mattn/go-sqlite3"
 )
 
+// WithTransactionalDbSession calls the callback with an session within a transaction
```

Comment:
Moved the transaction functions here. Now it is more noticeable that there are few of them with only small differences not sure why. Seems like the difference is only if the callback takes `Context` or `DBSession` param (and whether it is global function or method of SqlStore but that is for other bigger PR). Could we unify it and thus have only `InTransaction` and `InTransactionWithRetry`?
1. We could have `callback (ctx Context, sess *DbSession)` in both.
2. `callback (ctx Context)` in both with some `getSessionFromCtx()` function to make it less icky getting the session from context
3. `callback (sess *DbSession)` in both. Not exactly sure the context is needed here as it seems it is used only as container for the session nothing else. I did not see any cancelation based on the context or any other values there.

Reply:
We still need two functions for legacy reasons. (I guess we should mark the global one with Deprecated so its shows in peoples editors) but they should definitely both use the same underlying function. Trying to come up with a reason for not doing this.. I think I was working on this before going on parental leave and just didn't finish up. 

3) the context makes it possible for the caller to execute multiple `bus.Dispatch` in one transaction. So it's up to the caller to decide what goes into one transaction. at least that's the goal. 

Great to see some work in this area! :)

Reply:
A meant more in the sense that the context is not probably needed inside the callback function and so it could be unboxed before. But it seems to me now that having context with some getter function makes more sense as there could be other values in the future or timeouts. I would go with option 2 in that case, but probably as a separate PR in the future.